server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java

```diff
@@ -338,6 +341,55 @@ public void invalidAddressesShouldBeAllowed() throws Exception {
     }
 
     @Test
+    public void dateFromHeaderShouldBeUsedIfPresent() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Date: Wed, 17 May 2017 14:18:52 +0200\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZonedDateTime.of(2017, 05, 17, 14, 18, 52, 00, ZoneId.of("Europe/Paris")));
+    }
+
+    @Test
+    public void internalDateShouldBeUsedIfNoDateInHeaders() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZONED_DATE);
+    }
+
```

Comment:
In fact this will have no sens. This PR does not do better than old code considering timezone, and old sucked already.
There is too case 1) we have date header 2) we have no date hear:
1) We do technically have timezone information in the header (+XX not that I do not call that timezone) but this is lost in mime4j that return a date and not a ZonedTimeDate
2) We do not have date header than we don't have any clue of which timezone to use

In both case the exact moment is not lost and because we have to set a ZonedTimeZone but we do not know the timezone we use UTC. For me it's not a big deal, the date is mail not event, what matter is too know when the mail has been sent that all, we do not need to know in which timezone the mail has been written..
So there is two options (1) you agree with me (2) you think timezone matter:
1/ We should change the jmap/model/Message to use Date and ZonedDateTime because the time is lying on the fact that we know the timezone but we don't. I can do that on that PR
2/ In this case, I would argue this is out of the scope of PR. I would also ask you to tell me how you do when there is no mail header. How do you guess when you +XX which timezone it is (there is no good way to know actually).